pkg/providers/azureopenai/chat.go

```diff
@@ -123,23 +123,26 @@ func (c *Client) doChatRequest(ctx context.Context, payload *ChatRequest) (*sche
 		return nil, fmt.Errorf("failed to send azure openai chat request: %w", err)
 	}
 
-	defer resp.Body.Close() // TODO: handle this error
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		bodyBytes, err := io.ReadAll(resp.Body)
 		if err != nil {
 			c.telemetry.Logger.Error("failed to read azure openai chat response", zap.Error(err))
 		}
 
-		// TODO: Handle failure conditions
-		// TODO: return errors
 		c.telemetry.Logger.Error(
 			"azure openai chat request failed",
 			zap.Int("status_code", resp.StatusCode),
 			zap.String("response", string(bodyBytes)),
 			zap.Any("headers", resp.Header),
 		)
 
+		if resp.StatusCode == http.StatusTooManyRequests {
+			return nil, clients.NewRateLimitError(nil) // TODO: read real cooldown delay from headers
```

Comment:
Could we get the deadline when the provider becomes available from the response instead of nil?